File: 2021/11/25/build_array.py
"""
https://leetcode.com/problems/consecutive-characters/
Author: Andrew Jarombek
Date: 11/25/2021
"""

from typing import List


def build_array(target: List[int], n: int) -> List[str]:
    result = []

    index = 0
    i = 1

    while i <= n and index < len(target):
        if target[index] == i:
            result.append("Push")
            index += 1
        else:
            result.append("Push")
            result.append("Pop")

        i += 1

    return result

File: 2021/12/27/BuildArray.java
/**
 * https://leetcode.com/problems/build-array-from-permutation/
 * @author Andrew Jarombek
 * @since 12/27/2021
 */

public class BuildArray {
    public int[] buildArray(int[] nums) {
        int[] result = new int[nums.length];

        for (int i = 0; i < nums.length; i++) {
            result[i] = nums[nums[i]];
        }

        return result;
    }
}
